lib/pengine/native.c

```diff
@@ -572,7 +572,7 @@ pcmk__native_output_string(const pcmk_resource_t *rsc, const char *name,
     if (node && !(node->details->online) && node->details->unclean) {
         have_flags = add_output_flag(outstr, "UNCLEAN", have_flags);
     }
-    if (node && (node == rsc->lock_node)) {
+    if ((node != NULL) && pcmk__same_node(node, rsc->private->lock_node)) {
```

Comment:
`node != NULL` is redundant if we start using `pcmk__same_node`

Reply:
not if both are null

Reply:
If `pcmk__same_node()` returns true then `node1` is not NULL
```
static inline bool
pcmk__same_node(const pcmk_node_t *node1, const pcmk_node_t *node2)
{
    return (node1 != NULL) && (node2 != NULL)
           && (node1->details == node2->details);
}
```

Reply:
Ah I forgot that -- the real issue is that in a few places, the first node can be NULL and the second non-NULL, but the second doesn't matter unless the first is given.